File: day1/day1.go
package main

import (
	"bufio"
	"fmt"
	"strconv"
	"os"
)

func fuelRequired(mass int) int {
	fuel := (mass / 3) - 2
	if fuel < 0 {
		return 0
	}
	return fuel + fuelRequired(fuel)
}

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	sum := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		mass, err := strconv.Atoi(scanner.Text())
		if err != nil {
			fmt.Println(err)
		}
		sum += fuelRequired(mass)
	}
	fmt.Println(sum)
}

File: day1/day1.py
def fuel_required(mass):
    fuel = (mass // 3) - 2
    return 0 if fuel < 0 else fuel + fuel_required(fuel)


with open("input.txt") as f:
    masses = [int(line) for line in f]
    fuels = [fuel_required(mass) for mass in masses]
    total_fuel = sum(fuels)
    print(total_fuel)
